src/Xamarin.Android.Build.Tasks/Tasks/LinkAssembliesNoShrink.cs

```diff
@@ -53,10 +53,12 @@ public override bool RunTask ()
 				for (int i = 0; i < SourceFiles.Length; i++) {
 					var source = SourceFiles [i];
 					var destination = DestinationFiles [i];
+					var assemblyDefinition = resolver.GetAssembly (source.ItemSpec);
```

Comment:
It is possible to check `Profile.IsSdkAssembly (source.ItemSpec) || Profile.IsProductAssembly (source.ItemSpec)` and not load the assembly at all here? That might help perf slightly.